lib/rules/define-macros-order.js

```diff
@@ -0,0 +1,255 @@
+/**
+ * @author Eduard Deisling
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const utils = require('../utils')
+
+// ------------------------------------------------------------------------------
+// Helpers
+// ------------------------------------------------------------------------------
+
+const MACROS_EMITS = 'defineEmits'
+const MACROS_PROPS = 'defineProps'
+const ORDER = [MACROS_EMITS, MACROS_PROPS]
+const DEFAULT_ORDER = [MACROS_EMITS, MACROS_PROPS]
+
+/**
+ * Get an index of the first statement after imports and interfaces in order
+ * to place defineEmits and defineProps before this statement
+ * @param {Program} program
+ */
+function getTargetStatementPosition(program) {
+  const blockStatements = new Set([
+    'ImportDeclaration',
+    'TSInterfaceDeclaration',
+    'TSTypeAliasDeclaration',
+    'EmptyStatement'
+  ])
+
+  for (const [index, item] of program.body.entries()) {
+    if (!blockStatements.has(item.type)) {
+      return index
```

Comment:
I don't think most people will do that, but could you check directives like `"use strict"` just in case?